errors_test.go

```diff
@@ -273,3 +273,49 @@ func TestErrAPISetValue(t *testing.T) {
 	err := chunk.SetValue(1, 42, "hello")
 	testError(t, err, errAPI.Error(), columnCountErrMsg)
 }
+
+func TestGetDuckDBError(t *testing.T) {
```

Comment:
Using real errors is a good idea, but I must admit that some errors are hard to trigger, so it might be hard to cover all corner cases.
However I will change to test with the actual exceptions, thanks for the advice.

Reply:
Yes, I agree; many exceptions are hard to trigger. We can add a few tests to cover the error map (constraint exception, invalid input, (maybe) not implemented), and to cover `Is(err error)` and `Error`. Also, since we expose `DuckDBError`, we don't need an additional `struct` (like below) in the tests, no?
```Go
struct {
 		msg string
 		typ DuckDBErrorType
 	}
```

Reply:
That's right. I'd like to add a test that checks the actual exceptions and simplify the `TestGetDuckDBError` to focus on coner cases.